base/iobuffer.jl

```diff
@@ -80,13 +80,15 @@ function read_sub{T}(from::AbstractIOBuffer, a::AbstractArray{T}, offs, nel)
     return a
 end
 
-function read(from::AbstractIOBuffer, ::Type{UInt8})
+@inline function read(from::AbstractIOBuffer, ::Type{UInt8})
+    ptr = from.ptr
```

Comment:
This isn't really #9755; it is actually legal for LLVM to perform this optimization on its own, but it apparently won't unless you enable the `julia -O` passes.

#9755 is still relevant here, because even with all the optimizations we still have to store to `from.ptr` because it could alias `from.data`. Also because right now if you have a store in the same loop as `read` these optimizations probably don't help much :disappointed: